template/CHART_NAME/templates/daemonset.yaml

```diff
@@ -0,0 +1,126 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels: {{- include "common.labels.standard" . | nindent 4 }}
+    app.kubernetes.io/component: %%COMPONENT_NAME%%
+    {{- if .Values.commonLabels }}
+    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
+    {{- end }}
+  name: {{ template "common.names.fullname" . }}
+  {{- if .Values.commonAnnotations }}
+  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
+  {{- end }}
+spec:
+  {{- if .Values.%%MAIN_OBJECT_BLOCK%%.updateStrategy }}
+  updateStrategy: {{- toYaml .Values.%%MAIN_OBJECT_BLOCK%%.updateStrategy | nindent 4 }}
+  {{- end }}
+  selector:
+    matchLabels: {{- include "common.labels.matchLabels" . | nindent 6 }}
+      app.kubernetes.io/component: %%COMPONENT_NAME%%
+  template:
+    metadata:
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.podAnnotations }}
+      annotations: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.podAnnotations "context" $) | nindent 8 }}
+      {{- end }}
+      labels: {{- include "common.labels.standard" . | nindent 8 }}
+        app.kubernetes.io/component: %%COMPONENT_NAME%%
+        {{- if .Values.%%MAIN_OBJECT_BLOCK%%.podLabels }}
+        {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.podLabels "context" $) | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ template "%%TEMPLATE_NAME%%.serviceAccountName" . }}
+      {{- include "%%TEMPLATE_NAME%%.imagePullSecrets" . | nindent 6 }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.affinity }}
+      affinity: {{- include "common.tplvalues.render" ( dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.affinity "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.nodeSelector }}
+      nodeSelector: {{- include "common.tplvalues.render" ( dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.nodeSelector "context" $) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.tolerations }}
+      tolerations: {{- include "common.tplvalues.render" (dict "value" .Values.%%MAIN_OBJECT_BLOCK%%.tolerations "context" .) | nindent 8 }}
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.priorityClassName }}
+      priorityClassName: {{ .Values.%%MAIN_OBJECT_BLOCK%%.priorityClassName | quote }}
+      {{- end }}
+      {{- if .Values.%%MAIN_OBJECT_BLOCK%%.podSecurityContext.enabled }}
+      securityContext:
```

Comment:
It may make sense to use something like this https://github.com/bitnami/charts/pull/3615/files#diff-6946198ac609f7c34b1501979477ffb37b098aa16a5d4e79348fe344f1a2b042L42-R42